### PR TITLE
Add the Weight of Executing the Instruction to v2 Extrinsics

### PR DESCRIPTION
**Pull request by HenriqueNogara**
Adds the weight of executing the instruction to `add_and_affirm_instruction_with_memo_v2` and `add_instruction_with_memo_v2`.

### modified API

- Adds `execute_scheduled_instruction_v2` extrinsic;

### CODE REVIEW

**Review by adamdossa — APPROVED** [review]

File: pallets/settlement/src/lib.rs [hunk 1]
```diff
@@ -973,8 +1010,8 @@ decl_module! {
         }
 
         /// Root callable extrinsic, used as an internal call to execute a scheduled settlement instruction.
-        #[weight = <T as Config>::WeightInfo::execute_scheduled_instruction(*_legs_count)]
-        fn execute_scheduled_instruction(origin, id: InstructionId, _legs_count: u32) {
+        #[weight = <T as Config>::WeightInfo::execute_scheduled_instruction(*_fungible_transfers, *_nfts_transfers)]
+        fn execute_scheduled_instruction(origin, id: InstructionId, _fungible_transfers: u32, _nfts_transfers: u32) {
```
Comment (Neopallium) [inline pallets/settlement/src/lib.rs]: We can add the new parameter for 6.0, but not for 5.x.

I recommend moving the code to a `base_exec...` function that take the two counts, then add a new extrinsic that takes both counts, while leaving this one to only take a `leg_count` (just pass it as the fungible count).  This old extrinsic can be renamed `execute_scheduled_instruction_old` and the code below that schedules execution can use the new extrinsic. 

We need to make sure that there isn't any scheduled calls to the old extrinsic before removing/changing it.

**Review by Neopallium — CHANGES_REQUESTED** [review]

File: pallets/weights/src/pallet_settlement.rs [hunk 2]
```diff
@@ -338,42 +338,51 @@ impl pallet_settlement::WeightInfo for SubstrateWeight {
...
+    // Storage: Settlement VenueFiltering (r:101 w:0)
+    // Storage: Settlement InstructionLegStatus (r:101 w:101)
     // Storage: Portfolio PortfolioLockedAssets (r:1 w:1)
-    // Storage: Asset Frozen (r:1 w:0)
+    // Storage: Portfolio PortfolioLockedNFT (r:100 w:100)
+    // Storage: Asset Frozen (r:101 w:0)
     // Storage: Asset DisableInvestorUniqueness (r:1 w:0)
     // Storage: Timestamp Now (r:1 w:0)
     // Storage: Identity Claims (r:52 w:0)
-    // Storage: Portfolio Portfolios (r:2 w:0)
```
Comment (Neopallium) [inline pallets/weights/src/pallet_settlement.rs]: I wonder why these reads got removed.

**Review by Neopallium — APPROVED** [review]

File: pallets/weights/src/pallet_settlement.rs (hunk 2 above)
Comment (HenriqueNogara) [reply pallets/weights/src/pallet_settlement.rs]: Fixed by not using the default portfolio.

**Comment by Neopallium** [comment]
Please update the description.

**Review by adamdossa — APPROVED** [review]

**Pull request merged**